commit dd5ac3dcb22eb0f30cfa3137542fdec260bca0ef
Author: Wesley Chalmers <Wesley.Chalmers@amd.com>
Date:   Tue May 29 17:45:05 2018 -0400

    drm/amd/display: fix use of uninitialized memory
    
    [ Upstream commit f3e077d95ca0a016fdf3d6b1e97a9910dfdaff17 ]
    
    DML does not calculate chroma values for RQ when surface is not YUV, but DC
    will unconditionally use the uninitialized values for HW programming.
    This does not cause visual corruption since HW will ignore garbage chroma
    values when surface is not YUV, but causes presubmission tests to fail
    golden value comparison.
    
    Signed-off-by: Wesley Chalmers <Wesley.Chalmers@amd.com>
    Signed-off-by: Eryk Brol <eryk.brol@amd.com>
    Reviewed-by: Wenjing Liu <Wenjing.Liu@amd.com>
    Acked-by: Harry Wentland <harry.wentland@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/gpu/drm/amd/display/dc/dml/dml1_display_rq_dlg_calc.c b/drivers/gpu/drm/amd/display/dc/dml/dml1_display_rq_dlg_calc.c
index c2037daa8e66..0efbf411667a 100644
--- a/drivers/gpu/drm/amd/display/dc/dml/dml1_display_rq_dlg_calc.c
+++ b/drivers/gpu/drm/amd/display/dc/dml/dml1_display_rq_dlg_calc.c
@@ -239,6 +239,8 @@ void dml1_extract_rq_regs(
 	extract_rq_sizing_regs(mode_lib, &(rq_regs->rq_regs_l), rq_param.sizing.rq_l);
 	if (rq_param.yuv420)
 		extract_rq_sizing_regs(mode_lib, &(rq_regs->rq_regs_c), rq_param.sizing.rq_c);
+	else
+		memset(&(rq_regs->rq_regs_c), 0, sizeof(rq_regs->rq_regs_c));
 
 	rq_regs->rq_regs_l.swath_height = dml_log2(rq_param.dlg.rq_l.swath_height);
 	rq_regs->rq_regs_c.swath_height = dml_log2(rq_param.dlg.rq_c.swath_height);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1114_dd5ac3dcb22eb0f30cfa3137542fdec260bca0ef.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1114_dd5ac3dcb22eb0f30cfa3137542fdec260bca0ef.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
